spec/lib/models/i18n_repository_spec.rb

```diff
@@ -0,0 +1,168 @@
+      allow(i18n_provider).to receive(:update_file)
+
+      expect(i18n_provider).to receive(:add_directory_if_missing)
+      i18n_provider.upload_english(translation_file, path_to_translation, title)
+    end
+
+    context 'with the file already on Crowdin' do
+      before do
+        allow(i18n_provider).to receive(:file_exists_in_directory?).and_return(true)
+      end
+
+      it 'updates the file on Crowdin' do
+        allow(i18n_provider).to receive(:add_directory_if_missing)
+
+        expect(i18n_provider).to receive(:update_file)
+        i18n_provider.upload_english(translation_file, path_to_translation, title)
+      end
+    end
+
+    context 'with the file not already on Crowdin' do
+      before do
+        allow(i18n_provider).to receive(:file_exists_in_directory?).and_return(false)
+      end
+
+      it 'updates the file on Crowdin' do
+        allow(i18n_provider).to receive(:add_directory_if_missing)
+
+        expect(i18n_provider).to receive(:add_file)
+        i18n_provider.upload_english(translation_file, path_to_translation, title)
+      end
+    end
+  end
+
+  describe '#add_directory_if_missing' do
+    context 'with the directory missing on Crowdin' do
+      before do
+        allow(i18n_provider).to receive(:crowdin_directory_exists?).and_return(false)
+        i18n_provider.instance_variable_set(:@crowdin, crowdin)
+      end
+
+      it 'triggers the creation of the directory on Crowding' do
+        expect(crowdin).to receive(:add_directory).with(crowdin_directory)
+        i18n_provider.add_directory_if_missing
+      end
+    end
+  end
+
+  describe '#request_build' do
+    before do
+      i18n_provider.instance_variable_set(:@crowdin, crowdin)
+    end
+
+    it 'triggers the export of translations' do
+      expect(crowdin).to receive(:export_translations)
+      i18n_provider.request_build
+    end
+  end
+
+  describe '#download_locales' do
+    let(:path) { 'test-path' }
+
+    before do
+      i18n_provider.instance_variable_set(:@crowdin, crowdin)
+    end
+
+    it 'triggers the download to the correct path' do
+      expect(crowdin).to receive(:download_translation).with('all', output: path)
+      i18n_provider.download_locales(path)
+    end
+  end
+
+  describe '#translation_status_high_enough?' do
+    let(:code) { 'test-code' }
+    let(:percent) { 100 }
+    let(:translations_statuses) { [ { 'code' => code, 'translated_progress' => percent } ] }
+
+    before do
+      i18n_provider.instance_variable_set(:@translations_statuses, translations_statuses)
+    end
+
+    context 'with the translation status being to low' do
+      let(:acceptance_leve) { 101 }
+
+      it 'returns false' do
+        expect(i18n_provider.translation_status_high_enough?(code, acceptance_leve) ).to be_falsey
+      end
+    end
+
+    context 'with the translation status being high enough' do
+      let(:acceptance_leve) { 99 }
+
+      it 'returns false' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.